app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileActivity.java

```
@@ -0,0 +1,170 @@
+                            @Override
+                            public void onClick(final SweetAlertDialog sDialog) {
+                                sDialog.dismiss();
+                                final Intent intent = new Intent(CreateProfileActivity.this, LoginActivity.class);
+                                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+                                CreateProfileActivity.this.startActivity(intent);
+                            }
+                        })
+                        .show();
+            } catch (IOException e) {
+                new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                        .setTitleText(getString(R.string.could_not_create_profile))
+                        .setContentText(getString(R.string.could_not_create_profile_explanation))
+                        .setConfirmText(getString(R.string.dialog_ok))
+                        .setConfirmClickListener(null)
+                        .show();
+            }
+        } else {
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(getString(R.string.could_not_create_profile))
+                    .setContentText(getString(R.string.could_not_create_profile_explanation))
+                    .setConfirmText(getString(R.string.dialog_ok))
+                    .setConfirmClickListener(null)
+                    .show();
+        }
+    }
+
+    /**
+     * Generates a KeyPair based on the type selected.
+     *
+     * @return a {@link KeyPair} with the key type selected
+     */
+    private KeyPair generateKeyPair() {
+        final RadioButton rsaKeyButton = (RadioButton) findViewById(R.id.radioRSAKey);
+        if (rsaKeyButton.isChecked()) {
+            return serviceLocator.getKeyGenerator().generateRSAKeyPair("NervousFish generated key");
+        }
+        return null;
```

Can you throw an exception here instead of returning null?